src/Nest/Aggregations/AggregateDictionary.cs

```diff
@@ -121,30 +121,30 @@ public FiltersAggregate Filters(string key)
 
 		public GeoCentroidAggregate GeoCentroid(string key) => TryGet<GeoCentroidAggregate>(key);
 
-		public SignificantTermsAggregate SignificantTerms(string key)
+		public SignificantTermsAggregate<TKey> SignificantTerms<TKey>(string key)
 		{
 			var bucket = TryGet<BucketAggregate>(key);
 			return bucket == null
 				? null
-				: new SignificantTermsAggregate
+				: new SignificantTermsAggregate<TKey>
 				{
 					BgCount = bucket.BgCount,
 					DocCount = bucket.DocCount,
-					Buckets = bucket.Items.OfType<SignificantTermsBucket>().ToList(),
+					Buckets = GetSignificantTermsBuckets<TKey>(bucket.Items).ToList(),
 					Meta = bucket.Meta
 				};
 		}
 
-		public SignificantTermsAggregate SignificantText(string key)
+		public SignificantTermsAggregate<TKey> SignificantText<TKey>(string key)
```

Comment:
I think having a non generic overload defaulting to `TKey` as string would be a nicer experience for the most common case.

Reply:
I can add that.

Reply:
This is done.